client/executor/src/native_executor.rs

```diff
@@ -283,3 +339,34 @@ macro_rules! native_executor_instance {
 		}
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use runtime_interface::runtime_interface;
+
+	#[runtime_interface]
+	trait MyInterface {
+		fn say_hello_world(data: &str) {
+			println!("Hello world from: {}", data);
+		}
+	}
+
+	native_executor_instance!(
+		pub MyExecutor,
+		test_runtime::api::dispatch,
+		test_runtime::native_version,
+		(my_interface::HostFunctions, my_interface::HostFunctions),
```

Comment:
Shouldn't a name clash be an error?

Reply:
While implementing it, I was not sure. Maybe we want to support at one point that users even can overwrite the default runtime interface functions. 

(This is just on the host side, so the stuff that is registered in the executor)

Reply:
A feature we would welcome very much for more easy Intel SGX support. :)

Reply:
Which functions do you need to override and why?

Reply:
We had running an runtime in a trusted execution environment, (which implies `no_std` , but not wasm). With the previous implementation we just patched the sr-io module, which I hoped to circumvent now. 

But now I believe that this is not the core issue. Is it correct that in order to do that, one would need to implement a `no_std` compatible `Externalities` trait?

Reply:
Hmm?